### DISCUSSION TITLE
Synology NAS DS923+and Meshcentral - docker

### DISCUSSION THREAD

**Discussion by Nomex1337** [Ylianst/MeshCentral#5698] [Q&A]
Hello, 

i have issues with Clients not connecting to Meshcentral Server.

I have installed "Meshcentral" on my Synology NAS DS923+ using Docker.

In the Container Manager, Meshcentral is running as a container and has these port settings: "local port:8086, container port:443, protocol:tcp".

Furthermore, I have set up a DDNS: remoute.my-domain.com which points to my NAS. The address is secured with a Let's Encrypt certificate at https://remoute.my-domain.com/.

On the Synology NAS, I have set up a reverse proxy with the following settings:
Source-- Protocol: https, Hostname: remoute.my-domain.com Port: 443.
Destination-- Protocol: http, Hostname: 192.168.178.57, Port: 8086.

Websocket: 

Headername: Upgrade  value: $http_upgrade
Headername: Connection  value: $connection_upgrade

I can access the Meshcentral Server locally via http://192.168.178.57:8086/ and over the Internet via https://remoute.my-domain.com/ and https://remoute.my-domain.com:443

Now to the problem:

In Meshcentral, I can add a Mesh Agent. When I create this for Windows, download it, and then install it on the client PC, this client PC is not listed in Meshcentral. It seems that the client/agent cannot establish a connection.

The client/agent points to the server URL: wss://remoute.my-domain.com:443/agent.ashx. It seems that Meshcentral creates the agent with this address. However, the address should be wss://remoute.my-domain.com/agent.ashx.

Do I need to change something in the Meshcentral config, or open any other ports and set up the reverse proxy differently? Does the aliasPort need to be set up in the reverse proxy or router port forwarding?

Here is my config:
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "settings": {
    "plugins":{"enabled": false},
    "_mongoDb": null,
    "cert": "remoute.my-domain.com",
    "WANonly": false,
    "LANonly": false,
    "sessionKey": "T41N90ZIZ3WC5ZR0KRPPMF9AQ8FIR9DWHE0X4P847BT6STF3",
    "port": 443,
    "_aliasPort": 443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": true,
    "SelfUpdate": false,
    "AllowFraming": false,
    "WebRTC": false
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "minify": true,
      "NewAccounts": false,
      "localSessionRecording": false,
      "_userNameIsEmail": true,
      "_certUrl": ""
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "",
    "_names": "",
    "production": false
  }
}
```

I hope you guys can help me

**Comment by si458** [comment]
You need to do a few things.
1. set `trustedProxy` under settings to the ip address of ur reverse proxy
2. If the meshcentral isn't handling the tls (creating the ssl cert etc) you need to also set `tlsoffload` under settings to the ip of ur reverse proxy 
3. You then need to set `certUrl` which is under domains to the domain url of ur website

**Comment by Nomex1337** [comment]
hello si485 thanks for fast answer is this right ? but server is not more starting.

192.168.178.57 is my NAS IP

```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "settings": {
    "plugins":{"enabled": false},
    "_mongoDb": null,
    "cert": "remoute.my-domain.com",
    "WANonly": false,
    "LANonly": false,
    "sessionKey": "T41N90ZIZ3WC5ZR0KRPPMF9AQ8FIR9DWHE0X4P847BT6STF3",
    "trustedProxy": 192.168.178.57,
    "port": 443,
    "_aliasPort": 8443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": 192.168.178.57,
    "SelfUpdate": false,
    "AllowFraming": false,
    "WebRTC": false
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "minify": true,
      "NewAccounts": false,
      "localSessionRecording": false,
      "_userNameIsEmail": true,
      "certUrl": "remoute.my-domain.com"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "",
    "_names": "",
    "production": false
  }
}
```

**Reply by si458** [reply]
try this as you havent formatted it correctly, the ips need to be in speech marks and the url needs to be a full url inc port
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "settings": {
    "plugins":{"enabled": false},
    "_mongoDb": null,
    "cert": "remoute.my-domain.com",
    "WANonly": false,
    "LANonly": false,
    "sessionKey": "T41N90ZIZ3WC5ZR0KRPPMF9AQ8FIR9DWHE0X4P847BT6STF3",
    "trustedProxy": "192.168.178.57",
    "port": 443,
    "_aliasPort": 8443,
    "redirPort": 80,
    "_redirAliasPort": 80,
    "AgentPong": 300,
    "TLSOffload": "192.168.178.57",
    "SelfUpdate": false,
    "AllowFraming": false,
    "WebRTC": false
  },
  "domains": {
    "": {
      "_title": "MyServer",
      "_title2": "Servername",
      "minify": true,
      "NewAccounts": false,
      "localSessionRecording": false,
      "_userNameIsEmail": true,
      "certUrl": "https://remoute.my-domain.com:443"
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before>",
    "_email": "",
    "_names": "",
    "production": false
  }
}
```

**Reply by Nomex1337** [reply]
i was only adding "certUrl": "remoute.my-domain.com" and now i can see the client pc.


"trustedProxy": 192.168.178.57, "TLSOffload": 192.168.178.57, is not working server is then not starting. This was my fault with the "..." 

but no i need ?

"trustedProxy": "192.168.178.57",
"TLSOffload": "192.168.178.57",
"certUrl": "https://remoute.my-domain.com:443"

**Reply by si458** [reply]
the code actually adds the `https://` to `certurl` if you dont set it, just looked at code
`if (obj.config.domains[i].certurl.indexOf('://') < 0) { obj.config.domains[i].certurl = 'https://' + obj.config.domains[i].certurl; }`

the `trustedproxy` basically lists the correct IP address that gets passed by the proxy, otherwise connection ips might look like they all coming from ur reverse proxy, and thr tlsoffload, offloads all the SSL to your reverse proxy if thats where you are doing the ssl certificate
but again both of those NEED to be in string format, like `"trustedProxy": "192.168.178.57", "TLSOffload": "192.168.178.57"`

but if it works with just the certurl then leave as is!

**Comment by Nomex1337** [comment]
Thank you so much for your help si458 - have a good time
